### ISSUE TITLE
[test failure] Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#7085]
from https://travis-ci.org/elastic/logstash/jobs/231472539

```
  1) Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern
     Failure/Error: expect(reload_stats["successes"]).to eq(1)
     
       expected: 1
            got: 0
     
       (compared using ==)
     # ./specs/reload_config_spec.rb:68:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
Finished in 13 minutes 24 seconds (files took 5.48 seconds to load)
32 examples, 1 failure
Failed examples:
rspec ./specs/reload_config_spec.rb:29 # Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern
Randomized with seed 53928
```

**Comment by jsvd** [comment]
another failure in the same test, but in a different place: https://travis-ci.org/elastic/logstash/jobs/255092893#L3386-L3396

```
 1) Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern
     Failure/Error: send_data(initial_port, sample_data)
     
     Errno::ECONNREFUSED:
       Connection refused - connect(2) for "127.0.0.1" port 13941
     # ./framework/helpers.rb:33:in `send_data'
     # ./specs/reload_config_spec.rb:36:in `block in (root)'
     # /home/travis/.rvm/gems/jruby-9.1.10.0/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `block in (root)'
Finished in 22 minutes 52 seconds (files took 9.1 seconds to load)
31 examples, 1 failure
Failed examples:
rspec ./specs/reload_config_spec.rb:29 # Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern
Randomized with seed 60783
```

**Comment by jakelandis** [comment]
Another similar issue:
```
15:38:31   1) Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern
15:38:31      Failure/Error: send_data(reload_port, sample_data)
15:38:31      
15:38:31      Errno::ECONNREFUSED:
15:38:31        Connection refused - connect(2) for "127.0.0.1" port 10406
15:38:31      # ./framework/helpers.rb:33:in `send_data'
15:38:31      # ./specs/reload_config_spec.rb:55:in `block in (root)'
15:38:31 
15:38:31 Finished in 9 minutes 55 seconds (files took 6.2 seconds to load)
15:38:31 17 examples, 1 failure
15:38:31 
15:38:31 Failed examples:
15:38:31 
15:38:31 rspec ./specs/reload_config_spec.rb:29 # Test Logstash service when config reload is enabled can reload when changes are made to TCP port and grok pattern
15:38:31 
15:38:31 Randomized with seed 54368
```
in the log above:
```
15:29:12 [2017-11-14T15:29:12,776][FATAL][logstash.runner          ] An unexpected error occurred! {:error=>#<NoMethodError: undefined method `size' for nil:NilClass>, :backtrace=>["/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline_reporter.rb:90:in `block in worker_states'", "org/jruby/RubyArray.java:2486:in `map'", "org/jruby/RubyEnumerator.java:323:in `each'", "org/jruby/RubyEnumerator.java:501:in `with_index'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline_reporter.rb:88:in `worker_states'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline_reporter.rb:57:in `block in to_hash'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb:80:in `inflight_batches'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline_reporter.rb:56:in `to_hash'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline_reporter.rb:51:in `snapshot'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/shutdown_watcher.rb:88:in `pipeline_report_snapshot'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/shutdown_watcher.rb:63:in `block in start'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/2.3.0/gems/stud-0.0.23/lib/stud/interval.rb:20:in `interval'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/shutdown_watcher.rb:59:in `start'", "/opt/logstash/build/logstash-7.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/shutdown_watcher.rb:35:in `block in start'"]}
15:29:15 [2017-11-14T15:29:15,250][INFO ][logstash.javapipeline    ] Pipeline terminated {"pipeline.id"=>"main"}
15:29:15 [2017-11-14T15:29:15,869][INFO ][logstash.javapipeline    ] Starting pipeline {:pipeline_id=>"main", "pipeline.workers"=>16, "pipeline.batch.size"=>125, "pipeline.batch.delay"=>5, "pipeline.max_inflight"=>2000, :thread=>"#<Thread:0x72f1cfd6 run>"}
15:29:16 [2017-11-14T15:29:16,060][INFO ][logstash.javapipeline    ] Pipeline started {"pipeline.id"=>"main"}
15:29:16 [2017-11-14T15:29:16,161][INFO ][logstash.agent           ] Pipelines running {:count=>1, :pipelines=>["main"]}
15:30:09   can reload when changes are made to TCP port and grok pattern (FAILED - 1)
15:30:09 Tearing down services
```

**Comment by original-brownbear** [comment]
Fixed in #8669 :)
